### DISCUSSION TITLE
Google OAuth skips consent screen

### DISCUSSION THREAD

**Discussion by lucasnantonio** [supabase/supabase#3712] [Questions]
I've just implemented Google OAuth in my app, and I've noticed a strange behavior:

- If the user has 2 or more Google Accounts signed in, the expected behavior occurs: they're taken to the Google consent screen and can choose which account to use from there.

- However, if a user has exactly 1 google account signed in elsewhere (eg. on Gmail), the OAuth button will log them into the app immediately, without showing them the consent screen and giving them a chance to choose another account.

Could this be caused by the way I'm doing a redirect?

```  const handleLoginWithGoogle = async () => {
    try {
      setLoading(true);
      console.log(process.env.NEXT_PUBLIC_HOST);
      const { user, session, error } = await supabase.auth.signIn(
        {
          provider: "google",
        },
        { redirectTo: process.env.NEXT_PUBLIC_HOST }
      );
      if (error) throw error;
    } catch (error) {
      alert(error.error_description || error.message);
    } finally {
      setLoading(false);
    }
  };```

**Comment by silentworks** [comment]
This should show the consent screen the first time they do this, but subsequent logins would definitely not show the consent screen as this is how all social logins work.

**Reply by lucasnantonio** [reply]
I'm not sure this is necessarily the best experience. 

On Google's screen, in addition to choosing an already-connected account, users have a chance to add an account they haven't logged in with before. If this screen is skipped, they never get the chance to switch to a new account, for example, if they're in a new device or browser.

This is already handled well if the user has 2 or more accounts. Every time I log out of my app and try to login again, I see the screen:
![CleanShot 2021-11-01 at 10 47 51](https://user-images.githubusercontent.com/13240286/139690694-5dcf8745-2e51-4658-9019-d1c435934230.png)

But if I only have 1 account currently logged in with Google, this screen is skipped, and therefore I'm stuck with my first account. To fix it, I need to go to Gmail or another google app, and add the account there, which is counter intuitive IMO.

**Reply by madebydor** [reply]
Totally agree with this.
Is there anything to do about it?

**Reply by grschafer** [reply]
![image](https://user-images.githubusercontent.com/694225/143917168-886a1174-7123-4585-8a2a-f420e0cacc53.png)

silentworks is correct -- on first login you'll see a screen like the above. On subsesquent logins (if you only have 1 google account), it will immediately log you in without an intermediate screen. **If you want to see the consent screen again**, you need to revoke access to the oauth app in your google account's security settings (should be at a url like https://myaccount.google.com/permissions).

**Comment by mliakos** [comment]
It should **always** display the consent screen and simply don't ask for password, once the account is selected. The user might want to either use a different logged in account **or** add a new one. Both of those actions are currently impossible with the current behavioour, unless we clear certain cookies. End users should definitely not have to clear any cookies :) Plus, on PWAs it's plain impossible.

**Comment by timdobranski** [comment]
Did anyone ever find any official documentation for this behavior? I'm having a similar experience: I have multiple accounts, and on first login the consent screen appears and I choose my gmail account. But after signout and every signin after that clicking the login button skips the consent screen and signs the user in with the previous account immediately. It sounds just like what was described here, but the difference is that I do have multiple accounts and still not seeing the consent screen after first login. Users are essentially stuck with their first login choice forever, unless they sign out of the account on the browser. I can also confirm that after logout the local storage is cleared and I've even tried manually deleting the user from the auth users table after login but it still remembers and uses the original login. Here's the login code:

  const handleSignIn = async () => {
    const { data, error } = await supabase.auth.signInWithOAuth({
      provider: 'google',
    });
    if (error) {
      console.error(error);
    } else {
      console.log('User signed in:', data);
    }
  };

**Comment by timdobranski** [comment]
Nevermind, found the solution. For anyone else who stumbles across this, to force the option of choosing an account after every logout, pass the prompt: 'select_account' query param inside of options, queryParams:

  const handleSignIn = async () => {
    const { data, error } = await supabase.auth.signInWithOAuth({
      provider: 'google',
      options: {
        queryParams: {
          prompt: 'select_account'
        }
      }
